src/main/java/org/broadinstitute/hellbender/tools/spark/sv/QNameFinder.java

```diff
@@ -0,0 +1,43 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import org.apache.commons.collections4.iterators.SingletonIterator;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * Class to find the template names associated with reads in specified intervals.
+ */
+public final class QNameFinder implements Function<GATKRead, Iterator<QNameAndInterval>> {
+    private final ReadMetadata metadata;
+    private final List<SVInterval> intervals;
+    private final Iterator<QNameAndInterval> noName = Collections.emptyIterator();
+    private int intervalsIndex = 0;
```

Comment:
Each instance of QNameFinder is processing one partition of the BAM.  The reads are known to be coordinate ordered, as are the intervals.  We don't want to search from the beginning of the intervals list with each applied read -- especially for the later partitions we'd be walking a long way to find our spot.  Instead, we save our location from the last read, and increment when necessary.  Point is that because the data is ordered, we'll never need to back up in the intervals list, we'll only occasionally need to read further into the list.
It would work as an instance variable, but it would be much, much slower.